### ISSUE TITLE
I can send a modification command from the Go client

### ISSUE THREAD

**Issue by saig0** [camunda/camunda#9661]
## Description

We want to send a modification command for a process instance from the Go client. As a result, the Go client should send a gRPC message to the gateway and handle the response. Similar to the Java API: #9660 

* extend the Go API to send a modification command for a process instance
* required parameter: the key of the process instance
* optional parameters:
  * activate instruction(s): 
    * (required) the element id
    * (optional) the ancestor element instance key
    * (optional) variable instruction(s): 
      * variables - provided as the common formations
      * (optional) the element id of the variable scope
  * terminate instruction(s):
    * (required) the element instance key   

Blocked by 
- #9659

**Comment by lzgabel** [comment]
Hi @saig0, @korthout. Is there any progress on the current feature? If no one has contributed to this yet, you can assign the task to me.

**Comment by korthout** [comment]
Thanks @lzgabel. We'd be happy to receive a contribution for this

**Comment by ingorichtsmeier** [comment]
Once this done, it should be added to the `zbctl` implementation, too.

**Comment by korthout** [comment]
> Once this done, it should be added to the `zbctl` implementation, too.

@ingorichtsmeier Is there a specific need for this that you're aware of? Modifications may be complex (and verbose) to describe in the request. So, I'm wonder whether this is a suitable feature for `zbctl`.

**Comment by ingorichtsmeier** [comment]
Hi @korthout,  

for the sake of completeness. 

And to enable (manual) batch modifications with bash scripts. A customer requested batch modifications to go to production, and they are in Operate not yet available.

In reality, most times the modifications are easy: cancel one activity and start another one.

**Comment by korthout** [comment]
Thanks @ingorichtsmeier 

>In reality, most times the modifications are easy: cancel one activity and start another one.

So it sounds like we could even build this in tiny iterations, e.g.
1. allow very simple modifications, only specify `elementId` for activate instructions and `elementInstanceKey` for terminate instructions, i.e. required fields only ([grpc spec](https://docs.camunda.io/docs/apis-tools/grpc/#input-modifyprocessinstancerequest))
2. allow specifying the ancestor element key for activate instructions
3. allow specifying the variable instructions for activate instructions

@aleksander-dytko It might make sense to track this from product-hub perspective. And perhaps you can provide priority details using @ingorichtsmeier's explanation above.

**Comment by ingorichtsmeier** [comment]
Hi @aleksander-dytko, no, it's not blocking. 

I proposed a workaround with a BPMN Process invoking the Java API in a job worker.

**Comment by lzgabel** [comment]
Hi @korthout @megglos . Should this issue be transferred to the [@camunda-community-hub/zeebe-client-go](https://github.com/camunda-community-hub/zeebe-client-go) repository?

**Comment by korthout** [comment]
@lzgabel I sadly cannot transfer it, perhaps @megglos has permissions for this.

**Comment by nicpuppa** [comment]
@korthout @lzgabel GitHub doesn't offer the possibility to transfer issues to another organization, we have to clone it manually in this case 👍

**Comment by lzgabel** [comment]
> [@korthout](https://github.com/korthout) [@lzgabel](https://github.com/lzgabel) GitHub doesn't offer the possibility to transfer issues to another organization, we have to clone it manually in this case 👍

@nicpuppa FYI, I see that some issues are transferred to the new repository. Perhaps @megglos has permissions.
eg: camunda-community-hub/zeebe-client-go/issues/22

**Comment by korthout** [comment]
@megglos is out at the moment. If there's no rush let's wait for his return. Otherwise, feel free to open a new issue in that repo, and link back here.
